programs/uxd/src/instructions/register_alloyx_vault_depository.rs

```diff
@@ -0,0 +1,176 @@
+use crate::error::UxdError;
+use crate::events::RegisterAlloyxVaultDepositoryEvent;
+use crate::state::alloyx_vault_depository::ALLOYX_VAULT_DEPOSITORY_SPACE;
+use crate::state::AlloyxVaultDepository;
+use crate::utils::validate_collateral_mint_usdc;
+use crate::validate_is_program_frozen;
+use crate::Controller;
+use crate::ALLOYX_VAULT_DEPOSITORY_ACCOUNT_VERSION;
+use crate::ALLOYX_VAULT_DEPOSITORY_NAMESPACE;
+use crate::CONTROLLER_NAMESPACE;
+use anchor_lang::prelude::*;
+use anchor_spl::associated_token::AssociatedToken;
+use anchor_spl::token::Mint;
+use anchor_spl::token::Token;
+use anchor_spl::token::TokenAccount;
+
+#[derive(Accounts)]
+pub struct RegisterAlloyxVaultDepository<'info> {
+    /// #1
+    pub authority: Signer<'info>,
+
+    /// #2
+    #[account(mut)]
+    pub payer: Signer<'info>,
+
+    /// #3
+    #[account(
+        mut,
+        seeds = [CONTROLLER_NAMESPACE],
+        bump = controller.load()?.bump,
+        has_one = authority @UxdError::InvalidAuthority,
+    )]
+    pub controller: AccountLoader<'info, Controller>,
+
+    /// #4
+    #[account(
+        init,
+        seeds = [
+            ALLOYX_VAULT_DEPOSITORY_NAMESPACE,
+            alloyx_vault.key().as_ref(),
+            collateral_mint.key().as_ref()
```

Comment:
could add the controller here and stop registering depositories (the idiomatic way)

Reply:
But better follow what I started wrongly to keep things coherent

Reply:
Done! (in separate PR: https://github.com/UXDProtocol/uxd-program/pull/306)